code/obj/item/stun_baton.dm

```diff
@@ -419,6 +419,49 @@
 		logTheThing("combat", user, null, "swaps the power cell (<b>Cell type:</b> <i>[C.type]</i>) of [src] at [log_loc(user)].")
 		return
 
+	intent_switch_trigger(var/mob/user)
+		src.do_flip_stuff(user, user.a_intent)
+
+	attack_hand(var/mob/user)
+		if (src.flipped && user.a_intent != "harm")
+			user.show_text("You flip \the [src] the right way around as you grab it.")
+			src.flipped = false
+			src.update_icon()
+			user.update_inhands()
+		else if (user.a_intent == "harm")
+			src.do_flip_stuff(user, "harm")
+		..()
+
+	proc/do_flip_stuff(var/mob/user, var/intent)
+		if (intent == "harm")
+			if (src.flipped) //swapping hands triggers the intent switch too, so we dont wanna spam that
+				return
+			src.flipped = true
+			animate(src, transform = turn(matrix(), 120), time = 7 CENTI SECONDS) //turn partially
+			animate(transform = turn(matrix(), 240), time = 7 CENTI SECONDS) //turn the rest of the way
+			animate(transform = turn(matrix(), 180), time = 4 CENTI SECONDS) //finish up at the right spot
```

Comment:
In my opinion this is clearer as `0.07 SECONDS` etc.

Reply:
good call